### PR TITLE
Bg List Categories

### PR DESCRIPTION
**Pull request by cbgriffith**
# Description

Added a list of categories.  Created a GetAll method and used that pull all of the categories from the database and created a view off of that method into Views/Category/Index.cshtml.  I used the existing Post index.cshtml as a template in the views.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Testing Instructions

1. Navigate to the root directory for this repo (directory-to)/tabloidmvc-the-daily-debuggers
2. Start TabloidMVC.sln
3. Make sure your db is created with seed data
4. Run [01_TabloidMVC_Create_DB.sql](https://github.com/NewForce-Cohort-5/tabloidmvc-the-daily-debuggers/blob/main/SQL/01_TabloidMVC_Create_DB.sql) and [02_TabloidMVC_Seed_Data.sql](https://github.com/NewForce-Cohort-5/tabloidmvc-the-daily-debuggers/blob/main/SQL/02_TabloidMVC_Seed_Data.sql) queries in Visual Studio
5. Start the server
6. Log in as admin admin@example.com (if not already logged in as admin)
7. Manually navigate to categories by typing in /Category into the browser (I did not edit a link to categories yet in the nav) The url should look something like https://localhost:44336/Category
8. You should now see a list of all categories

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors
- [x] I have added test instructions that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by nicholasvperry** [comment]
I have tested the category list and it is working successfully.

File: TabloidMVC/Models/Category.cs [hunk 1]
```diff
@@ -1,8 +1,11 @@
-﻿namespace TabloidMVC.Models
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace TabloidMVC.Models
 {
     public class Category
     {
         public int Id { get; set; }
+        [Required]
```
Comment (andy-birt) [inline TabloidMVC/Models/Category.cs:8]: Nice use of annotation!

File: TabloidMVC/Repositories/CategoryRepository.cs [hunk 2]
```diff
@@ -1,4 +1,5 @@
 ﻿using System.Collections.Generic;
+using System.Linq;
```
Comment (andy-birt) [inline TabloidMVC/Repositories/CategoryRepository.cs:2]: We may not need this statement.

**Review by andy-birt — APPROVED** [review]
Works like a charm 🍀

**Comment by Tommy-Spurlock** [comment]
Not bad! However, it needs to include a SQL script. That way we can ensure whoever is running this to test has the appropriate data for both admin and categories. It would also be helpful to say what the login is for the admin account so it's obvious for the tester.

**Comment by Tommy-Spurlock** [comment]
# Much Better

Great edits. Don't be afraid to get explicit with those instructions. Remember we want to write our testing instructions so well that people who don't know code can still review your PR. 

This one's good to go. 

👍

**Pull request merged**